src/ILVerify/src/SimpleArrayOfTRuntimeInterfacesAlgorithm.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using Internal.IL;
+using Internal.TypeSystem;
+
+namespace ILVerify
+{
+    internal class SimpleArrayOfTRuntimeInterfacesAlgorithm : RuntimeInterfacesAlgorithm
+    {
+        private DefType[] _arrayRuntimeInterfaces;
+        private MetadataType[] _genericRuntimeInterfaces;
+        private ModuleDesc _systemModule;
+
+        public SimpleArrayOfTRuntimeInterfacesAlgorithm(ModuleDesc systemModule)
+        {
+            _systemModule = systemModule;
+            _arrayRuntimeInterfaces = _systemModule.GetKnownType("System", "Array").RuntimeInterfaces;
+            _genericRuntimeInterfaces = new MetadataType[]
+            {
+                _systemModule.GetKnownType("System.Collections.Generic", "IEnumerable`1"),
+                _systemModule.GetKnownType("System.Collections.Generic", "ICollection`1"),
+                _systemModule.GetKnownType("System.Collections.Generic", "IList`1")
```

Comment:
Good point. `IReadOnlyList<T>` requires `IReadOnlyCollection<T>`, so that interface gets forced in by the C# compiler. You'll need to add that part manually here, but CoreRT is fine.